### PR TITLE
[ci][py3.11] Build python wheels on mac os for 3.11

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
- Unifying the wheel building for macos since both could be done with conda. (We are already building wheels with conda for arm64) 
- Added support for 3.11 wheels which is then trivial. (The system installed python seems to have 3.11 missing on buildkite still). 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/build-wheel-macos.sh [hunk 1]
```diff
@@ -6,79 +6,66 @@ set -e
 # Show explicitly which commands are currently running.
```
Comment (rickyyx) [inline python/build-wheel-macos.sh:6]: This is actually moved from build-wheel-macos-arm64.sh

I think we could unify the two for building macos wheels.

**Comment by rickyyx** [comment]
Wheels built are here:
- arm64: https://buildkite.com/ray-project/oss-ci-build-pr/builds/24964#01889b3c-a9b8-4d2b-8fe1-724b035b3b08
- intel: https://buildkite.com/ray-project/oss-ci-build-pr/builds/24964#01889b3c-a9b5-49be-a9d3-14642f3f3496

**Review by krfricke — APPROVED** [review]
LGTM, thanks!

**Comment by can-anyscale** [comment]
I looked at this recently for docker image build and it seems that the latest miniconda does not have support for 3.11 yet. According to their website (https://docs.conda.io/en/main/miniconda.html), miniconda is still at 23.3.1; we need 23.5.0 for 3.11 support

**Comment by rickyyx** [comment]
> https://docs.conda.io/en/main/miniconda.html

I think when I installed a 23.3.1 - I was able to install py3.11 env with `conda install` though somehow.

**Comment by rickyyx** [comment]
> I looked at this recently for docker image build and it seems that the latest miniconda does not have support for 3.11 yet. According to their website (https://docs.conda.io/en/main/miniconda.html), miniconda is still at 23.3.1; we need 23.5.0 for 3.11 support

Actually, I think we are able to install 23.5.0 through CLI already. Let me add an update command just so that we are using the most recent ones.

**Pull request merged**